django/db/models/query.py

```diff
@@ -645,6 +645,19 @@ def create(self, **kwargs):
         Create a new object with the given kwargs, saving it to the database
         and returning the created object.
         """
+        non_model_fields = (
+            set(kwargs)
+            .difference(self.model._meta._property_names)
+            .difference(self.model._meta._many_to_one_field_names)
+            .difference(self.model._meta._concrete_field_names)
+        )
```

Comment:
We don't check so many cases in `save()`, do we need to do this here? :thinking: For example, `save()` with `update_fields` checks only `_non_pk_concrete_field_names`.

Maybe instead of checking all allowed fields, we should raise an exception when reverse one-to-one field name is in `kwargs` (`model._meta.related_objects`).

Reply:
That's a good idea, seems way simpler that way 😄